cli/command/utils.go

```diff
@@ -76,6 +77,69 @@ func PrettyPrint(i any) string {
 
 var ErrPromptTerminated = errdefs.Cancelled(errors.New("prompt terminated"))
 
+type promptOptions struct {
+	disableEcho bool
+}
+
+type PromptOptions func(*promptOptions)
+
+func WithHideUserInput(t bool) PromptOptions {
+	return func(p *promptOptions) {
+		p.disableEcho = t
+	}
+}
+
+// PromptForInput requests input from the user.
+//
+// If the user terminates the CLI with SIGINT or SIGTERM while the prompt is
+// active, the prompt will return an empty string ("") with an ErrPromptTerminated error.
+// When the prompt returns an error, the caller should propagate the error up
+// the stack and close the io.Reader used for the prompt which will prevent the
+// background goroutine from blocking indefinitely.
+func PromptForInput(ctx context.Context, ins *streams.In, outs *streams.Out, message string, opts ...PromptOptions) (string, error) {
+	options := promptOptions{}
+	for _, o := range opts {
+		o(&options)
+	}
+
+	_, _ = fmt.Fprint(outs, message)
+
+	if options.disableEcho {
+		oldState, err := term.SaveState(ins.FD())
+		if err != nil {
+			return "", err
+		}
+		_ = term.DisableEcho(ins.FD(), oldState)
+		defer func() {
+			_ = term.RestoreTerminal(ins.FD(), oldState)
+		}()
+	}
+
+	// On Windows, force the use of the regular OS stdin stream.
+	if runtime.GOOS == "windows" {
+		ins = streams.NewIn(os.Stdin)
+	}
+
+	result := make(chan string)
+
+	go func() {
+		reader := bufio.NewReader(ins)
+		line, err := reader.ReadString('\n')
+		if err != nil {
+			result <- ""
+		}
+		result <- strings.TrimSpace(line)
+	}()
```

Comment:
I think this goroutine will still be running after context is `Done`, and so it will still be reading from the stdin? 🤔

Reply:
I'll check quick and write a test

Reply:
fixed it, with `bufio.NewScanner` it should exit the goroutine when the user terminates the CLI (SIGTERM/SIGINT). We do the same with `PromptForConfirmation`

this is also why we return an error so that the caller can gracefully exit and close the input stream.